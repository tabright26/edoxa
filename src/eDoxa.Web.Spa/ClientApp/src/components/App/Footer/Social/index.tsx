import React, { FunctionComponent } from "react";
import {
  Facebook,
  Twitter,
  Discord,
  Linkedin,
  Instagram
} from "components/Shared/Social";

type Props = {
  className?: string;
};

export const Social: FunctionComponent<Props> = ({ className = null }) => (
  <ul className={`m-0 p-0 ${className}`}>
    <li className="d-inline mr-3">
      <Facebook size="lg" />
    </li>
    <li className="d-inline mx-3">
      <Twitter size="lg" />
    </li>
    <li className="d-inline mx-3">
      <Discord size="lg" />
    </li>
    <li className="d-inline mx-3">
      <Linkedin size="lg" />
    </li>
    <li className="d-inline ml-3">
      <Instagram size="lg" />
    </li>
  </ul>
);
