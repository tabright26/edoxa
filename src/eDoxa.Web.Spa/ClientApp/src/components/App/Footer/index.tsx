import React, { FunctionComponent } from "react";
import {
  getLegalTermsOfUsePath,
  getLegalPrivacyPolicyPath
} from "utils/coreui/constants";
import { Link } from "react-router-dom";

interface Props {
  className?: string;
}

const Footer: FunctionComponent<Props> = ({ className = null }) => (
  <div className={className}>
    &copy; {new Date(Date.now()).getFullYear()} eDoxa - All rights reserved.
    <span className="mx-2">|</span>
    <Link to={getLegalTermsOfUsePath()}>Terms of Use</Link>
    <span className="mx-2">|</span>
    <Link to={getLegalPrivacyPolicyPath()}>Privacy Policy</Link>
  </div>
);

export default Footer;
