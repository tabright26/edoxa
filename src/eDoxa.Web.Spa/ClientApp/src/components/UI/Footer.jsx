import React from 'react';
import { Navbar, Container } from 'react-bootstrap';

export default class Footer extends React.Component {
  render() {
    return (
      <footer>
        <Navbar fixed="bottom" bg="secondary" variant="dark">
          <Container>
            <Navbar.Text className="m-auto">
              <small>
                &copy; {new Date(Date.now()).getFullYear()} eDoxa - All rights
                reserved.
              </small>
            </Navbar.Text>
          </Container>
        </Navbar>
      </footer>
    );
  }
}
