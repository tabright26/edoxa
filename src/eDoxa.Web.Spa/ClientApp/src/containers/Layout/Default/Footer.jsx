import React from "react";
import { Link } from "react-router-dom";

const Footer = () => (
  <div className="ml-auto">
    &copy; {new Date(Date.now()).getFullYear()} eDoxa - All rights reserved.
    <span className="mx-2">|</span>
    <Link to="/terms-of-services">Terms of Services</Link>
  </div>
);

export default Footer;
