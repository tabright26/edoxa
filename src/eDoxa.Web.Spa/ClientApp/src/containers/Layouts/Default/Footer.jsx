import React, { Component } from "react";
import PropTypes from "prop-types";

const propTypes = {
  children: PropTypes.node
};

const defaultProps = {};

class Footer extends Component {
  render() {
    // eslint-disable-next-line
    const { children, ...attributes } = this.props;

    return (
      <React.Fragment>
        <span className="ml-auto">&copy; {new Date(Date.now()).getFullYear()} eDoxa - All rights reserved.</span>
      </React.Fragment>
    );
  }
}

Footer.propTypes = propTypes;
Footer.defaultProps = defaultProps;

export default Footer;
