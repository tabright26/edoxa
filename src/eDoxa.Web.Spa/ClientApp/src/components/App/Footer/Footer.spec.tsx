import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router";
import { Provider } from "react-redux";
import Footer from "./Footer";

it("renders without crashing", () => {
  //Arrange
  const store: any = {
    getState: () => {},
    dispatch: action => {},
    subscribe: () => {}
  };

  //Act
  const tree = renderer
    .create(
      <Provider store={store}>
        <MemoryRouter>
          <Footer />
        </MemoryRouter>
      </Provider>
    )
    .toJSON();

  //Assert
  expect(tree).toMatchSnapshot();
});
