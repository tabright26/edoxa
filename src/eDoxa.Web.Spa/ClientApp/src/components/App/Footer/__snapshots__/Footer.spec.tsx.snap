// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="ml-auto"
>
  Â© 
  2019
   eDoxa - All rights reserved.
  <span
    className="mx-2"
  >
    |
  </span>
  <a
    href="/terms-of-services"
    onClick={[Function]}
  >
    Terms of Services
  </a>
</div>
`;
